libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -208,18 +208,18 @@ void NameAndTypeResolver::warnVariablesNamedLikeInstructions() const
 	for (auto const& instruction: evmasm::c_instructions)
 	{
 		string const instructionName{boost::algorithm::to_lower_copy(instruction.first)};
-		auto declarations = nameFromCurrentScope(instructionName, true);
-		for (Declaration const* const declaration: declarations)
+		for (Declaration const* const declaration: nameFromCurrentScope(instructionName, true))
 		{
 			solAssert(!!declaration, "");
-			if (dynamic_cast<MagicVariableDeclaration const* const>(declaration))
-				// Don't warn the user for what the user did not.
-				continue;
```

Comment:
Actually I was about to push a PR which adds distinct errors for trying to access variables visible in Solidity but not in inline assembly. Well, only for the `MagicVariableDeclaration`, but was wondering if it should be extended.

This PR goes strictly against that notion so perhaps it would make sense briefly discussing this import resolution bit, it has so many (seemingly random) rules.

Reply:
Never mind! I thought this is the TypeChecker, that is what I am talking about :)